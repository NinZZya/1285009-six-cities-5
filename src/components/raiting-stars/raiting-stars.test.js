import React from 'react';
import renderer from 'react-test-renderer';
import RaitingStars from './raiting-stars';


const RAITING_TYPE = `place-card__stars`;

describe(`Should RaitingStars render correctly`, () => {
  it(`Should RaitingStars render correctly without type and rate 0`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            rate={0}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should RaitingStars render correctly without type and rate 2.5`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            rate={2.5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should RaitingStars render correctly without type and rate 5`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            rate={5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should RaitingStars render correctly with type and rate 0`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            type={RAITING_TYPE}
            rate={0}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should RaitingStars render correctly with type and rate 2.5`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            type={RAITING_TYPE}
            rate={2.5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should RaitingStars render correctly with type and rate 5`, () => {
    const tree = renderer
      .create(
          <RaitingStars
            type={RAITING_TYPE}
            rate={5}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
