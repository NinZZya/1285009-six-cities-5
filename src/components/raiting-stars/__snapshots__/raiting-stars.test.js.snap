// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should RaitingStars render correctly Should RaitingStars render correctly with type and rate 0 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should RaitingStars render correctly Should RaitingStars render correctly with type and rate 2.5 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "50%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should RaitingStars render correctly Should RaitingStars render correctly with type and rate 5 1`] = `
<div
  className="rating__stars place-card__stars"
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should RaitingStars render correctly Should RaitingStars render correctly without type and rate 0 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "0%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should RaitingStars render correctly Should RaitingStars render correctly without type and rate 2.5 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "50%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;

exports[`Should RaitingStars render correctly Should RaitingStars render correctly without type and rate 5 1`] = `
<div
  className="rating__stars undefined"
>
  <span
    style={
      Object {
        "width": "100%",
      }
    }
  />
  <span
    className="visually-hidden"
  >
    Rating
  </span>
</div>
`;
